dev/com.ibm.ws.webcontainer.servlet.3.1_fat.2/fat/src/com/ibm/ws/webcontainer/servlet31/fat/tests/CustomizedExceptionText.java

```diff
@@ -0,0 +1,126 @@
+     * Verify that the customized exception text configured in server.xml
+     * is returned instead of the default
+     * "SRVE0218E: Forbidden" message
+     * when the application sends a 403 Forbidden error.
+     */
+    @Test
+    public void tes403CustomMessage() throws Throwable {
+        server.stopServer();
+        server.setServerConfigurationFile("customizedExceptionTest/server-test-403-custom-message.xml");
+        server.startServer();
+        server.waitForStringInLogUsingMark("CWWKT0016I.*" + CUSTOM_ERROR_APP_NAME + ".*");
+
+        String requestURL = baseURL() + "/" + CUSTOM_ERROR_APP_NAME + "/forbidden";
+        String result = checkRequest(requestURL, 403, "Custom 403 Forbidden error message", "SRVE0218E");
+
+        assertTrue("Custom 403 text not found in response:\n" + result, result.contains("PASS"));
+    }
+
+    /**
+     * Verify that the customized exception text configured in server.xml
+     * is returned instead of the default
+     * "SRVE0232E: An exception occurred" message
+     * when a servlet triggers a 500 Internal Server Error.
+     */
+    @Test
+    public void test500CustomMessage() throws Throwable {
+        server.stopServer();
+        server.setServerConfigurationFile("customizedExceptionTest/server-test-500-custom-message.xml");
+        server.startServer();
+        server.waitForStringInLogUsingMark("CWWKT0016I.*" + CUSTOM_ERROR_APP_NAME + ".*");
+
+        String requestURL = baseURL() + "/" + CUSTOM_ERROR_APP_NAME + "/error";
+        String result = checkRequest(requestURL, 500, "Custom 500 internal server error message", "SRVE0232E");
+
+        assertTrue("Custom 500 text not found in response:\n" + result, result.contains("PASS"));
+    }
+
+    private String checkRequest(String URL, int responseCode, String expectedResText, String notExpectedResText) throws Throwable {
+        WebConversation wc = new WebConversation();
+        wc.setExceptionsThrownOnErrorStatus(false);
+        WebResponse resp = wc.getResponse(URL);
+
+        int code = resp.getResponseCode();
+        String body = resp.getText();
+
+        StringBuilder result = new StringBuilder();
+        result.append("Checking Response Code: Expected [" + responseCode + "], actual [" + code + "]. ");
+        result.append(code == responseCode ? "PASS\n" : "FAIL\n");
+
+        if (!expectedResText.equals("SKIP")) {
+            result.append("Checking Expected Response text: ");
+            result.append(body.contains(expectedResText) ? "PASS\n" : "FAIL\n");
+        }
+
+        if (!notExpectedResText.equals("SKIP")) {
+            result.append("Checking NOT Expected Response text: ");
+            result.append(body.contains(notExpectedResText) ? "FAIL\n" : "PASS\n");
+        }
+
+        LOGGER.log(Level.INFO, "checkRequest Body:\n" + body);
+        return result.toString();
+    }
+
+    private String baseURL() {
+        return "http://" + server.getHostname() + ":" + server.getHttpDefaultPort();
```

Comment:
Instead of doing this ourselves, why not leverage `public static URL createURL(LibertyServer server, String path) throws MalformedURLException` in `fattest.simplicity`?